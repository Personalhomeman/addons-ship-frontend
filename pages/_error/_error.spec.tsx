import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Error from '.';

describe('_error', () => {
  describe('renders ok', () => {
    [404, 500, 420, null].forEach(statusCode => {
      test(`when status code is ${statusCode}`, () => {
        const tree = shallowToJson(shallow(<Error statusCode={statusCode} />));
        expect(tree).toMatchSnapshot();
      });
    });
  });

  test('getInitialProps', () => {
    expect(Error.getInitialProps({ res: { statusCode: 123 }, err: { statusCode: 456 } } as any)).toEqual({
      statusCode: 123
    });

    expect(Error.getInitialProps({ err: { statusCode: 456 } } as any)).toEqual({
      statusCode: 456
    });

    expect(Error.getInitialProps({} as any)).toEqual({
      statusCode: null
    });
  });
});
