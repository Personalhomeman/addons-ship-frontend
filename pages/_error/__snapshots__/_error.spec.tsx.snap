// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_error renders ok when status code is 404 1`] = `
<Flex
  alignChildrenVertical="middle"
  backgroundColor="grape-5"
  color="aqua-1"
  direction="vertical"
  grow={true}
>
  <_default />
  <Flex
    paddingVertical="x10"
  >
    <AddonFooter
      addonName="Ship"
    />
  </Flex>
</Flex>
`;

exports[`_error renders ok when status code is 420 1`] = `
<Flex
  alignChildrenVertical="middle"
  backgroundColor="grape-5"
  color="aqua-1"
  direction="vertical"
  grow={true}
>
  <Flex
    alignChildrenVertical="middle"
    direction="vertical"
    grow={true}
  >
    <Text
      align="middle"
      config="5"
      margin="x8"
    >
      An unexpected error occured
    </Text>
  </Flex>
  <Flex
    paddingVertical="x10"
  >
    <AddonFooter
      addonName="Ship"
    />
  </Flex>
</Flex>
`;

exports[`_error renders ok when status code is 500 1`] = `
<Flex
  alignChildrenVertical="middle"
  backgroundColor="grape-5"
  color="aqua-1"
  direction="vertical"
  grow={true}
>
  <_default />
  <Flex
    paddingVertical="x10"
  >
    <AddonFooter
      addonName="Ship"
    />
  </Flex>
</Flex>
`;

exports[`_error renders ok when status code is null 1`] = `
<Flex
  alignChildrenVertical="middle"
  backgroundColor="grape-5"
  color="aqua-1"
  direction="vertical"
  grow={true}
>
  <Flex
    alignChildrenVertical="middle"
    direction="vertical"
    grow={true}
  >
    <Text
      align="middle"
      config="5"
      margin="x8"
    >
      An unexpected error occured
    </Text>
  </Flex>
  <Flex
    paddingVertical="x10"
  >
    <AddonFooter
      addonName="Ship"
    />
  </Flex>
</Flex>
`;
